protocols/protocol/cloudstate/event_sourced.proto

```diff
@@ -80,35 +123,72 @@ message EventSourcedReply {
     google.protobuf.Any snapshot = 5;
 }
 
-// Input message type for the gRPC stream in.
+//EventSourcedStreamIn TODO
+//
+//---
+//
+//`Specification`:
+//
+//  - An `EventSourcedStreamIn` is considered to be valid iff:
+//    - `message` TODO
+//      - `init` TODO
+//      - `event` TODO
+//      - `command` TODO
 message EventSourcedStreamIn {
+
+    // TODO
     oneof message {
+        // TODO
         EventSourcedInit init = 1;
+        // TODO
         EventSourcedEvent event = 2;
+        // TODO
         Command command = 3;
     }
 }
 
-// Output message type for the gRPC stream out.
+//EventSourcedStreamOut TODO
+//
+//---
+//
+//`Specification`:
+//
+//  - An `EventSourcedStreamOut` is considered to be valid iff:
+//    - `message` TODO
+//      - `reply` TODO
+//      - `failure` TODO
 message EventSourcedStreamOut {
+
+    // TODO
     oneof message {
+        // TODO
         EventSourcedReply reply = 1;
+        // TODO
         Failure failure = 2;
     }
 }
 
-// The Entity service
+//The *EventSourced* Service is the entrypoint for event sourced User Function entities.
+//
+// TODO: One stream will be established per active entity.
+// Once established, the first message sent will be Init, which contains the entity ID, and,
+// if the entity has previously persisted a snapshot, it will contain that snapshot. It will
+// then send zero to many event messages, one for each event previously persisted. The entity
+// is expected to apply these to its state in a deterministic fashion. Once all the events
+// are sent, one to many commands are sent, with new commands being sent as new requests for
+// the entity come in. The entity is expected to reply to each command with exactly one reply
+// message. The entity should reply in order, and any events that the entity requests to be
+// persisted the entity should handle itself, applying them to its own state, as if they had
+// arrived as events when the event stream was being replayed on load.
+//
```

Comment:
> The entity is expected to reply to each command with exactly one reply message.

What is the state of the event sourced entity after a failure was sent as a`EventSourcedStreamOut.Failure`?
Is this a fatal situation (not recoverable)? Should the entity instance be "passivated"...?

Reply:
@marcellanz Good question—what does the proxy do in this case right now? (Haven't got access to my codebase right now)

Reply:
@viktorklang yeah, could also check. The question is rather old and its here to mark a point that it could be described here for protocol semantics.

Reply:
@viktorklang may I describe here, in the spec, what the state of the entity is, according to the current proxy implementation?

Reply:
@marcellanz Yeah, that's a good idea, to provide a foundation for formalization. Thank you! :)